sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -65,6 +68,21 @@ public class ChatClient {
         )
 
         self.service = client.chat
+
+        var token: String?
+
+        credential.token(completionHandler: {(communicationAccessToken, _)
+            in
+            if communicationAccessToken != nil {
+                token = communicationAccessToken!.token
+            }
+        })
+
+        if token != nil {
+            self.signalingClient = getSignalingClient(token: token!)
```

Comment:
Why is this not simply an initializer? `SignalingClient(withToken: token)`?

Reply:
moved some logic to inside getSignalingClient